webapp/src/components/incoming_calls/hooks.ts

```diff
@@ -253,13 +260,25 @@ export const useOnChannelLinkClick = (call: IncomingCallNotification) => {
     if (global) {
         return () => {
             notificationSounds?.stopRing(); // User interacted with notifications, so stop ringing for _any_ incoming call.
+            dispatch(trackEvent(Telemetry.Event.NotificationClickGotoChannel, source));
             sendDesktopEvent('calls-link-click', {link: channelURL});
         };
     }
 
     return () => {
         notificationSounds?.stopRing();
+        dispatch(trackEvent(Telemetry.Event.NotificationClickGotoChannel, source));
         const win = window.opener ? window.opener : window;
         win.postMessage({type: 'browser-history-push-return', message: {pathName: channelURL}}, window.origin);
     };
 };
+
+export const telemetrySource = (onWidget: boolean) => {
+    if (onWidget) {
+        return Telemetry.Source.Widget;
+    } else if (window.location.pathname.includes(`${pluginId}/expanded/`)) {
```

Comment:
Is the usual `window.opener` check not valid here?

Reply:
I don't see why it wouldn't be... Changed.